src/AppInstallerCommonCore/AdminSettings.cpp

```diff
@@ -164,74 +226,128 @@ namespace AppInstaller::Settings
             out << YAML::Key << s_AdminSettingsYaml_BypassCertificatePinningForMicrosoftStore << YAML::Value << m_settingValues.BypassCertificatePinningForMicrosoftStore;
             out << YAML::Key << s_AdminSettingsYaml_InstallerHashOverride << YAML::Value << m_settingValues.InstallerHashOverride;
             out << YAML::Key << s_AdminSettingsYaml_LocalArchiveMalwareScanOverride << YAML::Value << m_settingValues.LocalArchiveMalwareScanOverride;
+            out << YAML::Key << s_AdminSettingsYaml_ProxyCommandLineOptions << YAML::Value << m_settingValues.ProxyCommandLineOptions;
+
+            if (m_settingValues.DefaultProxy)
+            {
+                out << YAML::Key << s_AdminSettingsYaml_DefaultProxy << YAML::Value << m_settingValues.DefaultProxy.value();
+            }
+
             out << YAML::EndMap;
 
             return m_settingStream.Set(out.str());
         }
+
+        auto GetPolicyStateForSetting(BoolAdminSetting setting)
+        {
+            auto policy = GetAdminSettingPolicy(setting);
+            return GroupPolicies().GetState(policy);
+        }
+
+        std::optional<std::reference_wrapper<const std::string>> GetPolicyStateForSetting(StringAdminSetting setting)
+        {
+            switch (setting)
+            {
+            case AppInstaller::Settings::StringAdminSetting::DefaultProxy:
+                return GroupPolicies().GetValueRef<ValuePolicy::DefaultProxy>();

```

Comment:
Yes, `GroupPolicies().GetValue*()` returns `std::nullopt` if the policy is not set.
Unless I'm misunderstanding what it does, the `reference_wrapper` allows us to return a reference of the value instead of copying it. That is the behavior I would expect if we were able to do `std::optional<std::string&>`.